### ISSUE TITLE
Mask mAP is not increasing, constant at 0

### ISSUE THREAD

**Issue by rahulg963** [dbolya/yolact#470]
I am trying to train on my custom data set with 8 classes.
This is my evaluation mAP result: 
```
Computing validation mAP (this may take a while)...
Calculating mAP...
       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box |  2.64 |  7.98 |  6.29 |  4.63 |  3.25 |  2.09 |  1.17 |  0.59 |  0.36 |  0.01 |  0.00 |
  mask |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |  0.00 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
```
After some more epochs, I can see my box mAP increasing, though mask mAP is always 0, I have checked it for training dataset as well as for validation dataset, after multiple epochs. Please see training logs below
```
[  2]    1690 || B: 4.251 | C: 2.751 | M: 265.478 | S: 0.343 | T: 272.823 || ETA: 34 days, 8:14:13 || timer: 0.904
[  2]    1700 || B: 4.342 | C: 2.771 | M: 270.877 | S: 0.335 | T: 278.325 || ETA: 34 days, 8:18:27 || timer: 0.950
[  2]    1710 || B: 4.437 | C: 2.782 | M: 277.467 | S: 0.343 | T: 285.030 || ETA: 34 days, 8:50:36 || timer: 0.883
[  2]    1720 || B: 4.523 | C: 2.805 | M: 278.029 | S: 0.338 | T: 285.695 || ETA: 34 days, 11:19:09 || timer: 0.869
```

**My Training dataset contains : 1150 images
My Validation dataset contains : 320 images
Images are of size : 1280 × 720**

Below snippet is from my config.py
```
my_custom_doc_dataset = dataset_base.copy({
    'name': 'my-data',
  
   'train_images': '/data/train/',
    'train_info':   '/data/train/coco_annotations.json',
  
    'valid_images':  '/data/val/',
    'valid_info':  '/data/val/coco_annotations.json',

    'has_gt': True,

    'class_names': ('floor', 'wall', 'left_wall', 'right_wall', 'ceiling', 'box', 'bag', 'conveyor_belt'),
    'label_map': {1: 1,2: 2,3: 3,4: 4,5: 5,6: 6,7: 7, 8: 8}
})

yolact_resnet101_doc_config = yolact_base_config.copy({
    'name': 'yolact_plus_resnet101_doc',
    'dataset': my_custom_doc_dataset,
    'num_classes': len(my_custom_doc_dataset.class_names) + 1,

    'max_size': 550,
})
```

**Comment by ic** [comment]
From your trace, the mask loss remains very high. Perhaps a problem with the mask RLE values in your dataset. Have you made sure these values are correct (using pycocotools) ?

I used to have similar values on the mask loss, leading to zeros on the mask mAP for ages. The root cause was the RLE encoding of the masks. After fixing the encoding, the mask loss decreases right from the beginning, getting positive mAP.

**Comment by rahulg963** [comment]
@ic I was able to solve this by resizing my train and validation images to 550 x 550 instead of original 1280 x 720.
Thank you for looking into this.

**Comment by ic** [comment]
Rahul, great if it works as expected. Your solution is a bit surprising, though. Yolact resizes automatically images and masks (I did not track the whole process, e.g. [here](https://github.com/dbolya/yolact/blob/master/data/coco.py#L219)). My images are from a FullHD camera, and I use the 550x550 `max_size` option in the configuration file. I could confirm the `DataLoader` gets all datum shaped well to 550x550 automatically.

I guess if your end result works, well, all good ! A bit mysterious at present, though!

**Comment by rahulg963** [comment]
It's mysterious for me as well. Though didn't get a chance as of now to go and check resizing functions of Yolact myself.

**Comment by DonKannalie** [comment]
@ic how would I check the mask RLE values with pycocotools? I also get 0 mAP for my masks. I used the VIA tool for annotation creation and exported them to COCO format (which is faulty; so now I distrust the annotated masks. They however seem fine when loading and visualizing them with pycocotools (loadAnns/showAnns)).

**Comment by DonKannalie** [comment]
Or rather, do I need to convert the annotation polygons to RLE?
